ModTek/Patches.cs

```diff
@@ -29,11 +80,30 @@ static void Postfix(VersionManifest __result)
             foreach (var entryKVP in ModTek.NewManifestEntries)
             {
                 var id = entryKVP.Key;
-                var path = entryKVP.Value.Path;
-                var type = entryKVP.Value.Type;
+                var newEntry = entryKVP.Value;
+                
+                if (newEntry.ShouldMergeJSON && __result.Contains(id, newEntry.Type))
+                {
+                    // read the manifest pointed entry and hash the contents
+                    DoJSONMerge.JSONHashes.Add(File.ReadAllText(__result.Get(id, newEntry.Type).FilePath).GetHashCode());
+
+                    // The manifest already contains this information, so we need to queue it to be merged
+                    var partialJSON = File.ReadAllText(newEntry.Path);
```

Comment:
Is this reading the file twice?

Reply:
No, that's reading two separate files, the source file that we're merging onto and the merge itself.